### PR TITLE
Use the folder object directly instead of looking it up by its name

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

File: src/gui/socketapi/socketuploadjob.cpp [hunk 1]
```diff
@@ -73,6 +73,10 @@ void SocketUploadJob::prepareTag(const AccountPtr &account)
 
 void SocketUploadJob::start()
 {
+    fail("needs porting");
```
Comment (fmoc) [inline src/gui/socketapi/socketuploadjob.cpp:76]: Note for self: PR is still WIP.

File: test/modeltests/testprotocolmodel.cpp [hunk 2]
```diff
@@ -38,11 +38,11 @@ private Q_SLOTS:
...
         QVERIFY(d.mkdir("bar"));
-        const QString foo = dir.path() + QStringLiteral("/foo");
-        const QString bar = dir.path() + QStringLiteral("/bar");
-        QVERIFY(TestUtils::folderMan()->addFolder(newAccountState, TestUtils::createDummyFolderDefinition(newAccountState->account(), foo)));
-        QVERIFY(TestUtils::folderMan()->addFolder(newAccountState, TestUtils::createDummyFolderDefinition(newAccountState->account(), bar)));
+        auto foo = TestUtils::folderMan()->addFolder(newAccountState, TestUtils::createDummyFolderDefinition(newAccountState->account(), dir.path() + QStringLiteral("/foo")));
+        auto bar = TestUtils::folderMan()->addFolder(newAccountState, TestUtils::createDummyFolderDefinition(newAccountState->account(), dir.path() + QStringLiteral("/bar")));
 
+        QVERIFY(foo);
+        QVERIFY(bar);
```
Comment (fmoc) [inline test/modeltests/testprotocolmodel.cpp:45]:
```suggestion
        QVERIFY(foo != nullptr);
        QVERIFY(bar != nullptr);
```

**Review by fmoc — CHANGES_REQUESTED** [review]

File: src/gui/folderman.h [hunk 3]
```diff
@@ -138,14 +138,8 @@ class FolderMan : public QObject
...
-    /** Returns the folder by alias or NULL if no folder with the alias exists. */
-    Folder *folder(const QString &);
-
-    /**
-     * Migrate accounts from owncloud < 2.0
-     * Creates a folder for a specific configuration, identified by alias.
-     */
-    Folder *setupFolderFromOldConfigFile(const QString &, AccountStatePtr account);
+    /** Returns the folder by id or NULL if no folder with the id exists. */
+    [[deprecated("directly reference the folder")]] Folder *folder(const QByteArray &id);
```
[the remark this answers is not in the document]
Comment (TheOneRing) [reply src/gui/folderman.h:142]: there is one last user... and I don't plan to touch that code in this review

File: src/gui/socketapi/socketuploadjob.cpp (hunk 1 above)
Comment (TheOneRing) [reply src/gui/socketapi/socketuploadjob.cpp:76]: nope I created another issue for this, this code is unused an appears to be broken for a long time now.
https://github.com/owncloud/client/issues/9585
Marked as p2 so we don't move it to 2.12 or something like that.

File: src/gui/accountsettings.cpp [hunk 4]
```diff
@@ -735,55 +723,52 @@ void AccountSettings::slotEnableCurrentFolder(bool terminate)
...
 }
 
 void AccountSettings::slotForceSyncCurrentFolder()
 {
-    FolderMan *folderMan = FolderMan::instance();
-    if (auto selectedFolder = folderMan->folder(selectedFolderAlias())) {
+    if (auto selectedFolder = this->selectedFolder()) {
         // Terminate and reschedule any running sync
-        for (auto f : folderMan->map()) {
+        for (auto f : FolderMan::instance()->folders()) {
```
[the remark this answers is not in the document]
Comment (TheOneRing) [reply src/gui/accountsettings.cpp:761]: Nope its a loop?

**Review by fmoc — APPROVED** [review]

**Pull request merged**